Assets/Scripts/NpcBehaviors/BaseNpc.cs

```diff
@@ -0,0 +1,272 @@
+/******************************************************************
+*    Author: Nick Grinstead
+*    Contributors: 
+*    Date Created: 5/17/24
+*    Description: Base NPC class for specific NPCs to derive from. 
+*       Has basic functionality for switching states and interacting
+*       that child scripts will expand upon.
+*******************************************************************/
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+using UnityEngine.InputSystem;
+
+public abstract class BaseNpc : MonoBehaviour
+{
+    /// <summary>
+    /// Structure used to store a collection of data related to the different states
+    /// Example: A StateDataGroup<Animation> could hold idle animations for each state
+    /// </summary>
+    /// <typeparam name="T">Type of data to be stored</typeparam>
+    [System.Serializable]
+    protected struct StateDataGroup<T>
+    {
+        [SerializeField] T _idleState;
```

Comment:
I know that these are inherently private by default, do we need to be more explicit? No idea what the standard requires, but consider adding a private.